ImageFinder/View/SearchImageViewController.swift

```diff
@@ -108,44 +159,83 @@ extension SearchImageViewController: UICollectionViewDataSource {
     print(indexPath.row)
     let urlString = resultImages[indexPath.row].thumbnailUrl
 
-    if let url = URL(string: urlString) {
-      if let data = try? Data(contentsOf: url) {
-        let image = UIImage(data: data)
-        print(data)
-        cell.thumbnailView.image = image
-        print(urlString)
+    DispatchQueue.global().async {
+      if let url = URL(string: urlString) {
+        if let data = try? Data(contentsOf: url) {
+          let image = UIImage(data: data)
+          DispatchQueue.main.async {
+            cell.thumbnailView.image = image
+          }
+        }
       }
     }
-//    cell.thumbnailView.image = UIImage(named: "jjong")
     return cell
   }
 }
 
 extension SearchImageViewController: UISearchBarDelegate {
-  func searchBarTextDidEndEditing(_ searchBar: UISearchBar) {
-    print("heyhey")
-    let time = DispatchTime.now() + .seconds(1)
-    let keyword = searchBar.text ?? ""
-
-    resultCollectionView.isHidden = true
-    searchLodingIndicator.startAnimating()
-
-    DispatchQueue.main.asyncAfter(deadline: time) {
-      self.searchLodingIndicator.stopAnimating()
-      self.resultCollectionView.isHidden = false
-      self.presenter.imageSearchService.doSearchImage(keyword: keyword)
-      self.resultCollectionView.reloadData()
+  func searchBarTextDidBeginEditing(_ searchBar: UISearchBar) {
+    setStatusView(status: .keywordTyping)
+  }
+
+  func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
+    searchDelayer.invalidate()
+    self.searchLoadingIndicator.startAnimating()
+    resultImages = []
+    page = 1
+
+    if true {
+      searchDelayer = Timer.scheduledTimer(timeInterval: 1, target: self, selector: #selector(self.doDelayedSearch), userInfo: searchText, repeats: false)
```

Comment:
```suggestion
      searchDelayer = Timer.scheduledTimer(
          timeInterval: 1,
          target: self,
          selector: #selector(self.doDelayedSearch),
          userInfo: searchText,
          repeats: false
      )
```
함수의 길이가 길어진다면 매개변수 기준으로 개행 시켜주세요.